src/geff/file_reader.py

```diff
@@ -0,0 +1,217 @@
+from pathlib import Path
+
+import numpy as np
+import zarr
+from numpy.typing import NDArray
+
+from . import utils
+from .dict_representation import GraphDict, PropDictNpArray, PropDictZArray
+from .metadata_schema import GeffMetadata
+
+
+class FileReader:
```

Comment:
Total nitpick but I would prefer "GeffReader." FileReader is very generic. And I think changing it is now or never 😆